src/auth/auth.service.ts

```diff
@@ -90,4 +91,37 @@ export class AuthService {
     await this.tokenService.revokeToken(userData.sub);
     return;
   }
+
+  // 소셜 로그인
+  async socialLogin(socialUser: SocialLoginPayload): Promise<TokenDto> {
+    const userNickname = await this.userService.findSocialUser(socialUser.nickname);
+    if (userNickname && userNickname.nickname !== socialUser.nickname) {
```

Comment:
```suggestion
    if (userNickname) {
```
위에서 nickname으로 찾았는데 nickname이 맞지 않을 때를 조건에 넣은게 맞는지 확인 부탁드려요.

개인적인 생각으로는 oauth에서의 닉네임은 충분히 겹칠 수 있어서 유니크값으로 닉네임을 운영할거라면 서비스만의 닉네임을 따로 받는게 좋아보이긴합니다.

Reply:
감사합니다! 저도 처음에는 nickname에 unique키를 설정해두었기 때문에 중복 여부까지 확인했었습니다. 그런데 닉네임이 겹칠 확률이 높다고 판단하여 nickname의 unique 제약을 제거하기로 했습니다. 이부분은 #42 이슈에서 해결할 예정입니다!